crates/hyperdrive-math/src/lp.rs

```diff
@@ -1149,4 +1946,162 @@ mod tests {
         }
         Ok(())
     }
+
+    #[tokio::test]
+    async fn fuzz_test_should_short_circuit_distribute_excess_idle_share_proceeds() -> Result<()> {
+        let chain = TestChain::new().await?;
+        let mock = chain.mock_lp_math();
+
+        // Fuzz the rust and solidity implementations against each other.
+        let mut rng = thread_rng();
+
+        for _ in 0..*FAST_FUZZ_RUNS {
+            // Generate random states.
+            let present_state = rng.gen::<State>();
+
+            let starting_present_value = rng.gen_range(fixed!(0)..=fixed!(1e24));
+            let present_value = rng.gen_range(fixed!(0)..=fixed!(1e24));
+            let active_lp_total_supply = rng.gen_range(fixed!(0)..=fixed!(1e24));
+            let lp_total_supply = rng.gen_range(fixed!(0)..=fixed!(1e24));
+
+            // Calculate the result from the Rust implementation.
+            let actual = present_state.should_short_circuit_distribute_excess_idle_share_proceeds(
+                active_lp_total_supply,
+                starting_present_value,
+                lp_total_supply,
+                present_value,
+            );
+
+            // Gather the parameters for the solidity call.
+            let mut params = DistributeExcessIdleParams::default();
+            params.active_lp_total_supply = active_lp_total_supply.into();
+            params.starting_present_value = starting_present_value.into();
+
+            // Make the solidity call and compare to the Rust implementation.
+            match mock
+                .should_short_circuit_distribute_excess_idle_share_proceeds(
+                    params,
+                    lp_total_supply.into(),
+                    present_value.into(),
+                )
+                .call()
+                .await
+            {
+                Ok(expected) => {
+                    if expected == true {
+                        assert!(actual? == true);
+                    } else {
+                        assert!(actual? == false);
+                    }
+                }
+                Err(_) => {
+                    assert!(actual.is_err())
+                }
+            }
+        }
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn fuzz_test_calculate_distribute_excess_idle_share_proceeds_net_long_edge_case(
+    ) -> Result<()> {
+        let chain = TestChain::new().await?;
+        let alice = chain.alice().await?;
+        let mock = chain.mock_lp_math();
+
+        // Fuzz the rust and solidity implementations against each other.
+        let mut rng = thread_rng();
+
+        for _ in 0..*FAST_FUZZ_RUNS {
+            // Generate random states.
+            let original_state = rng.gen::<State>();
+            let mut present_state = rng.gen::<State>();
+
+            //**************Get Variables*****************//
```

Comment:
this is new 👀 we don't use this type of comment to differentiate logic blocks anywhere else, probably best not to start here.